### ISSUE TITLE
FileVisitor::classDescriptionBuilder could be uninitialized (null)

### ISSUE THREAD

**Issue by ricfio** [phparkitect/arkitect#198]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | no
| Library Version | 0.2.4 
| PHP version     | 8.0.12

#### Summary

PHPArkitect throws the following error in its code base when checking my private project .

`Parse Error: Call to a member function addDependency() on null#0 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/Analyzer/FileVisitor.php(69): Arkitect\Analyzer\FileVisitor->addParamDependency()`

I have seen in the source code that:
- The error is thrown on `$this->classDescriptionBuilder->addDependency` in the FileVisitor::addParamDependency method (it might be useful to add an assertion like 'null !== $this->classDescriptionBuilder' ?)
- The `classDescriptionBuilder` property is not always initialized because all assignments are inside conditions, so when all conditions are false the property `classDescriptionBuilder` will not be initialized.

@AlessandroMinoccheri:
This issue is around the source code will be updated merging your PR in main branch, so I don't know if already solved by your PR:
https://github.com/phparkitect/arkitect/pull/197

#### Current behavior

![image](https://user-images.githubusercontent.com/69871782/142725842-a842da01-3088-4018-845b-3f0d9c1b9bbe.png)
```
PHPArkitect 0.2.4

Config file: phparkitect.php

analyze class set /home/ricfio/projects/portfolio-symfony/src

  0/99 [>---------------------------]   0%Parse Error: Call to a member function addDependency() on null#0 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/Analyzer/FileVisitor.php(69): Arkitect\Analyzer\FileVisitor->addParamDependency()
#1 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(200): Arkitect\Analyzer\FileVisitor->enterNode()
...
```
#### How to reproduce

I got the error checking my private project.
I don't know how to reproduce it, at this moment.

#### Expected behavior

No error in PHPArkitect after to have initialized classDescriptionBuilder in any case.

**Comment by AlessandroMinoccheri** [comment]
Do you have the same error if you use the actual official version?

**Comment by ricfio** [comment]
I'm already use the actual official version (0.2.4).
I have only observed that your PR will update the same methods of this issue.

**Comment by ricfio** [comment]
Here the full output of the stack:
```
PHPArkitect 0.2.4

Config file: phparkitect.php

analyze class set /home/ricfio/projects/portfolio-symfony/src

  0/99 [>---------------------------]   0%Parse Error: Call to a member function addDependency() on null#0 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/Analyzer/FileVisitor.php(69): Arkitect\Analyzer\FileVisitor->addParamDependency()
#1 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(200): Arkitect\Analyzer\FileVisitor->enterNode()
#2 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#3 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#4 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#5 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#6 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#7 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#8 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(91): PhpParser\NodeTraverser->traverseArray()
#9 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/Analyzer/FileParser.php(74): PhpParser\NodeTraverser->traverse()
#10 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/CLI/Runner.php(52): Arkitect\Analyzer\FileParser->parse()
#11 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/CLI/Runner.php(35): Arkitect\CLI\Runner->check()
#12 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/CLI/Check.php(75): Arkitect\CLI\Runner->run()
#13 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Command/Command.php(299): Arkitect\CLI\Check->execute()
#14 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Application.php(978): Symfony\Component\Console\Command\Command->run()
#15 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Application.php(295): Symfony\Component\Console\Application->doRunCommand()
#16 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun()
#17 /home/ricfio/projects/portfolio-symfony/vendor/bin/phparkitect(21) : eval()'d code(35): Symfony\Component\Console\Application->run()
#18 /home/ricfio/projects/portfolio-symfony/vendor/bin/phparkitect(21): eval()
#19 {main}Parse Error: Call to a member function addDependency() on null#0 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/Analyzer/FileVisitor.php(69): Arkitect\Analyzer\FileVisitor->addParamDependency()
#1 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(200): Arkitect\Analyzer\FileVisitor->enterNode()
#2 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#3 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#4 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#5 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#6 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#7 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#8 /home/ricfio/projects/portfolio-symfony/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(91): PhpParser\NodeTraverser->traverseArray()
#9 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/Analyzer/FileParser.php(74): PhpParser\NodeTraverser->traverse()
#10 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/CLI/Runner.php(52): Arkitect\Analyzer\FileParser->parse()
#11 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/CLI/Runner.php(35): Arkitect\CLI\Runner->check()
#12 /home/ricfio/projects/portfolio-symfony/vendor/phparkitect/phparkitect/src/CLI/Check.php(75): Arkitect\CLI\Runner->run()
#13 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Command/Command.php(299): Arkitect\CLI\Check->execute()
#14 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Application.php(978): Symfony\Component\Console\Command\Command->run()
#15 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Application.php(295): Symfony\Component\Console\Application->doRunCommand()
#16 /home/ricfio/projects/portfolio-symfony/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun()
#17 /home/ricfio/projects/portfolio-symfony/vendor/bin/phparkitect(21) : eval()'d code(35): Symfony\Component\Console\Application->run()
#18 /home/ricfio/projects/portfolio-symfony/vendor/bin/phparkitect(21): eval()
 99/99 [============================] 100%
```

**Comment by AlessandroMinoccheri** [comment]
It's important to understand if the code it's broken with new code or not,  please try to. use the official release to understand if the problem could be introduced with the new code :)

**Comment by ricfio** [comment]
I have the issue with the official 0.2.4 version, so what official version you would suggest me to use, please?

**Comment by AlessandroMinoccheri** [comment]
Ok, the version is correct. 
Now we need to understand which file generates the error.

**Comment by ricfio** [comment]
It's no so simple to isolate the file that generates the error because in the my project I have too errors due the self bug fixed by the PR:
https://github.com/phparkitect/arkitect/pull/197

I could have more info testing with a new PHPArkitect version after the PR merging into main branch.

**Comment by micheleorselli** [comment]
Would you mind trying running the check command with `-vvv`? That could give us additional hints

**Comment by ricfio** [comment]
Bingo! 
Now we known which file to watch... here it is:
`Application/Command/QuoteCommandInterface.php`

```
<?php

declare(strict_types=1);

namespace App\Application\Command;

use App\Domain\Quote\Quote;

interface QuoteCommandInterface
{
    /**
     * Save a stock quote.
     */
    public function save(Quote $quote): void;
}
```

I hope that is will be useful

**Comment by AlessandroMinoccheri** [comment]
Fixed by #203
